cli/command/registry/login.go

```diff
@@ -41,7 +41,7 @@ func NewLoginCommand(dockerCli command.Cli) *cobra.Command {
 	flags := cmd.Flags()
 
 	flags.StringVarP(&opts.user, "username", "u", "", "Username")
-	flags.StringVarP(&opts.password, "password", "p", "", "Password")
+	flags.StringVarP(&opts.password, "password", "p", "", "Password (deprecated, use --password-stdin)")
```

Comment:
It really depends on what's being deprecated, e.g. "deprecate `docker run`" in favour of `docker container run`" won't be something that could be done in three releases; some features cannot be removed.

Removing the `--email` flag from `docker login` (even though it was non-functional for more than a year, and had been printing a deprecation message for all that time) was already problematic